sys/include/net/gcoap.h

```diff
@@ -449,13 +449,37 @@ extern "C" {
 #define GCOAP_RESEND_BUFS_MAX      (1)
 #endif
 
+/**
+ * @name Bitwise positional flags for encoding resource links
+ * @{
+ */
+#define COAP_LINK_FLAG_SEPARATOR  (1)  /**< include link separator */
```

Comment:
I am wary of giving the local encoding function too much responsibility. We could use something like COAP_LINK_FLAG_INIT_RESLIST. This would indicate the need to do whatever is required to initialize a list of links, without specifically indicating 'first'. I think this would cover exclusion of the link separator, as well as some kind of header.